contracts/cw-proposal-single/src/v1_state.rs

```diff
@@ -0,0 +1,46 @@
+//! Helper methods for migrating from v1 to v2 state. These will need
+//! to be updated when we bump our CosmWasm version for v2.
+
+use voting::{
+    status::Status,
+    threshold::{PercentageThreshold, Threshold},
+    voting::Votes,
+};
+
+pub fn v1_percentage_threshold_to_v2(v1: voting_v1::PercentageThreshold) -> PercentageThreshold {
+    match v1 {
+        voting_v1::PercentageThreshold::Majority {} => PercentageThreshold::Majority {},
+        voting_v1::PercentageThreshold::Percent(p) => PercentageThreshold::Percent(p),
+    }
+}
+
+pub fn v1_threshold_to_v2(v1: voting_v1::Threshold) -> Threshold {
```

Comment:
can you explain again why we need this?

Reply:
this is used for migrations. rust doesn't guarantee that two types have the same in-memory layout despite having the same fields so we need to manually convert between the two types. if we could guarantee they had the same in-memory layout we could use `std::mem::transmute`. i strongly suspect this would work, just need some way to be sure (perhaps a test?).